src/hooks/useOverlayManager.tsx

```diff
@@ -32,8 +34,9 @@ export const OverlayManagerProvider: React.FC = ({ children }) => {
   return (
     <>
       <Global styles={overlayManagerStyles(scrollbarGap)} />
-      <OverlayManagerContext.Provider value={{ overlayOpen, setOverlayOpen: handleOverlayOpen }}>
+      <OverlayManagerContext.Provider value={{ overlayOpen, setOverlayOpen: handleOverlayOpen, overlayContainerRef }}>
         {children}
+        <div ref={overlayContainerRef}></div>
```

Comment:
This will need some styling to make sure it's not included in the regular page layout. We will need at least:
```css
position: fixed;
top: 0; /* right, bottom, left */
visibility: hidden; /* will need to switch to `visible` when `overlayOpen` */
```

Reply:
Also, if we include `z-index` here, the backdrop/dialog won't have to include it themselves